R/checkDeps.R

```diff
@@ -26,18 +27,32 @@ checkDeps <- function(descriptionFile = ".",
                       dependencyTypes = c("Depends", "Imports", "LinkingTo"),
                       action = "stop") {
   stopifnot(all(dependencyTypes %in% c("Depends", "Imports", "LinkingTo", "Suggests", "Enhances")))
-  stopifnot(action %in% c("stop", "warn", "pass"))
+  stopifnot(action %in% c("stop", "warn", "pass", "ask"))
   allDeps <- desc::desc_get_deps(descriptionFile)
   deps <- allDeps[allDeps$type %in% dependencyTypes, ]
   requirementMet <- Map(checkRequirement, package = deps$package, version = deps$version)
 
-  if (!all("TRUE" == requirementMet)) {
-    message <- paste(requirementMet[which("TRUE" != requirementMet)],
-                     collapse = "\n  ")
-    if ("stop" == action) {
-      stop(message)
-    } else if ("warn" == action) {
-      warning(message)
+  if (!all(requirementMet == "TRUE")) {
+    missing <- requirementMet[requirementMet != "TRUE"]
+    msg <- paste(missing, collapse = "\n  ")
+    if (action == "ask" && any(grepl(": ", missing))) {
+      if (!requireNamespace("renv", quietly = TRUE) || is.null(renv::project())) {
+        stop("'action = ask' is only supported in an activated renv.")
```

Comment:
```suggestion
        stop(msg)
```
I think, if we are not in a renv, doing `action = "stop"` instead of complaining is more useful